target-riscv/cpu.c

```diff
@@ -96,16 +92,7 @@ static void riscv_cpu_disas_set_info(CPUState *s, disassemble_info *info) {
 static void riscv_cpu_realizefn(DeviceState *dev, Error **errp)
 {
     CPUState *cs = CPU(dev);
-    RISCVCPU *cpu = RISCV_CPU(dev);
     RISCVCPUClass *mcc = RISCV_CPU_GET_CLASS(dev);
-    CPURISCVState *env = &cpu->env;
-
-    /* Enable GC ISA */
-    set_feature(env, RISCV_FEATURE_RVM);
```

Comment:
@asb I'm not sure I understand your objection?  qemu uses tb_flags for [the aarch64/aarch32 change](https://github.com/riscv/riscv-qemu/blob/master/target-arm/cpu.h#L2327) and a good number of other things, because tb_flags is the mechanism qemu uses to handle anything that (a) affects translations (b) might differ between CPUs in a configuration.

We could make `misa` writes no-op, but (a) the old behavior of accepting misa writes, letting them be read-back, but not actually honoring them was wrong, (b) I'd eventually like to support everything we architecturally can.

(I will later be sending a patch that merges riscv64-softmmu and riscv32-softmmu and adds MXL/SXL/UXL support, for instance.)